Note that additional work is required to enable compile
and execute feature in Solaris, e.g., solaris_infcall_munmap
method is needed and additional links are required to make
*-pc-solaris2.11-* executables and libcc1 library visible...

The patch will be sent to upstream.

--- gdb-15.1/gdb/osabi.c
+++ gdb-15.1/gdb/osabi.c
@@ -63,7 +63,7 @@ { "none", NULL },
 
   { "SVR4", NULL },
   { "GNU/Hurd", NULL },
-  { "Solaris", NULL },
+  { "Solaris", "solaris[2-9]\.[0-9][0-9]" },
   { "GNU/Linux", "linux(-gnu[^-]*)?" },
   { "FreeBSD", NULL },
   { "NetBSD", NULL },
